packages/shared/src/messages/MessageProxy.ts

```diff
@@ -78,12 +78,23 @@ export class RpcExtension {
           body: result,
           status: 'success',
         } as IMessageResponse);
-      } catch (e) {
+      } catch (err: unknown) {
+        let errorMessage: string;
+        // Depending on the object throw we try to extract the error message
+        if (err instanceof Error) {
+          errorMessage = err.message;
+        } else if (typeof err === 'string') {
+          errorMessage = err;
+        } else {
+          errorMessage = String(err);
+        }
+
         await this.webview.postMessage({
           id: message.id,
           channel: message.channel,
           body: undefined,
-          error: `Something went wrong on channel ${message.channel}: ${String(e)}`,
+          status: 'error',
+          error: errorMessage,
```

Comment:
![image](https://github.com/projectatomic/ai-studio/assets/42176370/4a18a7ce-bd80-46ee-ab3e-5a767d3a81dc)

No, because `Error` object in JS are not serializable

Reply:
And the traces are not relevant for the front. In an API communication, the message is important, so we can ignore the problems that occurs in the server